extension-app/src/components/Sidebar.jsx

```diff
@@ -1,7 +1,148 @@
-import React from 'react'
+import { useState } from "react";
+import { FiMenu } from "react-icons/fi";
+import { useSettings } from "../contexts/settings-context.js";
+import {
+  imageCategories,
+  refreshTimeOptions,
+  textColorOptions,
+} from "../utils/sidebar-filters.js";
+
+const Sidebar = () => {
+  const [showSidebar, setShowSidebar] = useState(false);
+  const {
+    settings: { imageCategory, imageRefreshTime, quoteRefreshTime },
+    setSettings,
+  } = useSettings();
+
+  const setCategory = (event) => {
+    setSettings((prev) => ({
+      ...prev,
+      imageCategory: event.target.value,
+    }));
+  };
+
+  const setImageRefreshTime = (event) => {
+    setSettings((prev) => ({
+      ...prev,
+      imageRefreshTime: Number(event.target.value),
+    }));
+  };
+  const setQuoteRefreshTime = (event) => {
+    setSettings((prev) => ({
+      ...prev,
+      quoteRefreshTime: Number(event.target.value),
+    }));
+  };
+
+  const setTextColor = (event) => {
+    setSettings((prev) => ({ ...prev, textColor: event.target.value }));
+  };
+
+  const resetHandler = () => {
+    setSettings({
+      imageCategory: "landscapes",
+      imageRefreshTime: 1800000,
+      quoteRefreshTime: 1800000,
+      textColor: "",
+    });
+    localStorage.clear();
+    window.location.reload();
+  };
 
-export const Sidebar = () => {
   return (
-    <div>Sidebar</div>
-  )
-}
+    <section className="sidebar-wrapper">
+      <FiMenu
+        size={25}
+        className="text-light"
+        onClick={() => setShowSidebar(!showSidebar)}
+      />
+
+      {showSidebar && (
+        <aside className="sidebar-container br-m px-xs py-s">
+          <div>
+            <div className="flex justify-content-space-bet">
+              <p className="text-light font-xbold mx-xs">Wallpaper</p>
+              <button className="btn btn-outline" onClick={resetHandler}>
+                Reset
+              </button>
+            </div>
+            {/* Wallpaper settings */}
```

Comment:
You can remove commented code after it's no longer needed

Reply:
will keep that in mind.